[WIP] Log4JXmlEventLayout using XmlLayout with InnerXml

ana1250 opened the pull request:
[WIP] Log4JXmlEventLayout using XmlLayout with InnerXml

ana1250 committed:
Removed Log4j renderer
Mapped the Chainsaw to log4j layout
Removed depenecy of log4j paramter class from System.xml
Removed depenecy of log4j layoutclass from System.xml
Fixed unit test for log4j paramter (had extra space).
Added Cdata property in XmlElement class.
Temp Unit test log4jxmllayout tests change

ana1250 committed:
Removed redundant xml helper class from targets.network
Added Invalid Xml Character Remover
Abstarcted logic of inner renderer into xml helper methods.
Added tests for cdata
Marked Previous test case for bad xml as obsolete
Added new unit test for invalid xml values.

snakefoot commented on src/NLog/Internal/XmlHelper.cs:
```diff
@@ -445,5 +445,77 @@ private static string EnsureDecimalPlace(string text)
+
+            return false;
+        }
+        private static bool IsValidXmlChar(int codePoint)
```
Notice there are also similar methods `XmlHelper.RemoveInvalidXmlChars`  + `XmlConvert.IsXmlChar`. Maybe there can be some code-reuse ?

snakefoot commented on src/NLog/LayoutRenderers/Wrappers/XmlEncodeLayoutRendererWrapper.cs:
```diff
@@ -57,6 +57,17 @@ public sealed class XmlEncodeLayoutRendererWrapper : WrapperLayoutRendererBase
+        /// <summary>
+        /// Remove Imvalid Xml Characters from inner
+        /// </summary>
+        public bool RemoveInvalidXml { get; set; } = true;
```
Can this be merged together with `XmlEncode`-property? Only when XmlEncode = false, then it should be raw-append without validation.

snakefoot commented on src/NLog/LayoutRenderers/Wrappers/XmlEncodeLayoutRendererWrapper.cs:
```diff
@@ -57,6 +57,17 @@ public sealed class XmlEncodeLayoutRendererWrapper : WrapperLayoutRendererBase
+        /// <summary>
+        /// Indicates whether the rendered value should be wrapped in <![CDATA[ ... ]]> section.
+        /// </summary>
+        public bool WrapInCData { get; set; } = false;
```
Can this renamed to `CDataEncode` ? (To match `XmlEncode`-property)

ana1250 committed:
Enhance Log4JXmlEventLayout XML handling

Updated Log4JXmlEventLayout to improve XML formatting and rendering. Key changes include:
- Added `ThrowableEncode` property for XML escaping control.
- Simplified `ScopeNestedSeparator` handling.
- Renamed `WrapInCData` to `CDataEncode`.
- Removed deprecated methods from `XmlHelper`.
- Updated test suite with new tests and removed obsolete ones.

snakefoot commented on src/NLog/Layouts/XML/XmlElement.cs:
```diff
@@ -102,12 +102,12 @@ public bool Encode
         /// </summary>
         public bool WrapValueInCData
```
Can this also be renamed to `CDataEncode` ?

ana1250 replied:
Fixed!

snakefoot commented on src/NLog.Targets.Network/Layouts/Log4JXmlEventLayout.cs:
```diff
@@ -45,7 +45,7 @@ namespace NLog.Layouts
     /// <remarks>
     /// <para>
-    /// This layout is not meant to be used explicitly. Instead you can use ${log4jxmlevent} layout renderer.
+    /// This layout is not meant to be used explicitly. Instead you can use ${log4jxmlevent} layout 
```
Think this remark should be removed entirely, since NLog supports both config-API and loading from config-file.

ana1250 replied:
Yes missed this,
thanks!

ana1250 committed:
Some minor fixes

ana1250 committed:
Minor Fix
